### PR TITLE
contrib: add gitian build support for github pull request

### PR DESCRIPTION
**Pull request by ken2812221**
- Add `-p` or `--pull` to specify the version is github pull request number

- `./gitian-build.py -bpDn test 13765` should work. This is helpful for someone want to try cross compiling.

### CODE REVIEW

**Review by kallewoof — COMMENTED** [review]
utACK bbf64d53374edb1be5dc7bd016369276addf3f56

File: contrib/gitian-build.py [hunk 1]
```diff
@@ -196,13 +197,21 @@ def main():
...
         setup()
 
     os.chdir('bitcoin')
+    if args.pull:
+        subprocess.check_call(['git', 'fetch', args.url, 'refs/pull/'+args.version+'/merge'])
+        os.chdir('../gitian-builder/inputs/bitcoin')
+        subprocess.check_call(['git', 'fetch', args.url, 'refs/pull/'+args.version+'/merge'])
+        args.commit = subprocess.check_output(['git', 'show', '-s', '--format=%H', 'FETCH_HEAD'], universal_newlines=True).strip()
+        args.version = 'pull-' + args.version
+    print(args.commit)
```
Comment (laanwj) [inline contrib/gitian-build.py]: is this intended or a left-over debug statement?

Comment (ken2812221) [reply contrib/gitian-build.py]: That's intended, move from the above one.

**Comment by laanwj** [comment]
utACK 46f83453701fc6116d97de6a16986388ba63488f in that case

**Pull request merged**